crates/ruff_python_formatter/src/expression/expr_unary_op.rs

```diff
@@ -1,17 +1,70 @@
-use crate::comments::Comments;
+use crate::comments::{trailing_comments, Comments};
 use crate::expression::parentheses::{
     default_expression_needs_parentheses, NeedsParentheses, Parentheses, Parenthesize,
 };
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::ExprUnaryOp;
+use crate::prelude::*;
+use crate::trivia::{SimpleTokenizer, TokenKind};
+use crate::FormatNodeRule;
+use ruff_formatter::FormatContext;
+use ruff_python_ast::prelude::UnaryOp;
+use ruff_text_size::{TextLen, TextRange};
+use rustpython_parser::ast::{ExprUnaryOp, Ranged};
 
 #[derive(Default)]
 pub struct FormatExprUnaryOp;
 
 impl FormatNodeRule<ExprUnaryOp> for FormatExprUnaryOp {
     fn fmt_fields(&self, item: &ExprUnaryOp, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let ExprUnaryOp {
+            range: _,
+            op,
+            operand,
+        } = item;
+
+        let operator = match op {
+            UnaryOp::Invert => "~",
+            UnaryOp::Not => "not",
+            UnaryOp::UAdd => "+",
+            UnaryOp::USub => "-",
+        };
+
+        text(operator).fmt(f)?;
+
+        let comments = f.context().comments().clone();
+
+        // Split off the comments that follow after the operator and format them as trailing comments.
+        // ```python
+        // (not # comment
+        //      a)
+        // ```
+        let leading_operand_comments = comments.leading_comments(operand.as_ref());
+        let trailing_operator_comments_end =
+            leading_operand_comments.partition_point(|p| p.position().is_end_of_line());
+        let (trailing_operator_comments, leading_operand_comments) =
+            leading_operand_comments.split_at(trailing_operator_comments_end);
+
+        if !trailing_operator_comments.is_empty() {
+            trailing_comments(trailing_operator_comments).fmt(f)?;
+        }
+
+        if op.is_not() {}
```

Comment:
huh?

Reply:
Whoops. I'm surprised that clippy isn't complaining about this.